app/components/pages/Studio.vue.ts

```diff
@@ -42,6 +44,16 @@ export default class Studio extends Vue {
     }, 1000);
   }
 
+  onDropHandler(event: DragEvent) {
+    const files = event.dataTransfer.files;
+
+    let fi = files.length;
+    while (fi--) {
```

Comment:
Since when we create new sources we put them on the top of the sources list, we have to add files with reversed direction to preserve the right order.